### PR TITLE
SONAR-8341 add sq started log

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java [hunk 1]
```diff
@@ -73,23 +87,207 @@ public LoggerContextListener enableJulChangePropagation(LoggerContext loggerCont
...
+    return configureRootLogLevel(newLevel);
+  }
+
+  /**
+   * Configure the log level of the root logger to the specified level.
+   *
+   * @throws IllegalArgumentException if the specified level is not one of {@link #ALLOWED_ROOT_LOG_LEVELS}
+   */
+  public Level configureRootLogLevel(Level newLevel) {
+    Logger rootLogger = getRootContext().getLogger(ROOT_LOGGER_NAME);
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java]: MAJOR Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

File: server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java [hunk 2]
```diff
@@ -73,23 +87,207 @@ public LoggerContextListener enableJulChangePropagation(LoggerContext loggerCont
...
+   *     the max number of files defined in property {@link #MAX_FILES_PROPERTY}</li>
+   *     <li>the logs will follow the specified log pattern</li>
+   *   </ul>
+   * </p>
+   *
+   * @see #buildLogPattern(RootLoggerConfig)
+   */
+  public FileAppender<ILoggingEvent> configureGlobalFileLog(Props props, RootLoggerConfig config, String logPattern) {
+    LoggerContext ctx = getRootContext();
+    Logger rootLogger = ctx.getLogger(ROOT_LOGGER_NAME);
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java]: MAJOR Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java [hunk 3]
```diff
@@ -254,18 +259,50 @@ public boolean isAllLogsToConsoleEnabled(Props props) {
...
    * @throws IllegalArgumentException if the value of the specified property is not one of {@link #ALLOWED_ROOT_LOG_LEVELS}
    */
-  public Level configureRootLogLevel(Props props, String propertyKey) {
-    Level newLevel = Level.toLevel(props.value(propertyKey, Level.INFO.toString()), Level.INFO);
-    return configureRootLogLevel(newLevel);
+  public Level configureRootLogLevel(Props props, String... propertyKeys) {
+    return configureLoggerLogLevel(getRootContext().getLogger(ROOT_LOGGER_NAME), props, propertyKeys);
+  }
+
+  private Level configureLoggerLogLevel(Logger logger, Props props, String... propertyKeys) {
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java]: MAJOR Make "configureLoggerLogLevel" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java [hunk 4]
```diff
@@ -254,18 +259,50 @@ public boolean isAllLogsToConsoleEnabled(Props props) {
...
+    if (!isAllowed(level)) {
+      throw new IllegalArgumentException(String.format("log level %s in property %s is not a supported value (allowed levels are %s)",
+        level, propertyKey, Arrays.toString(ALLOWED_ROOT_LOG_LEVELS)));
+    }
+    return level;
+  }
+
+  private static boolean isAllowed(Level level) {
+    for (Level allowedRootLogLevel : ALLOWED_ROOT_LOG_LEVELS) {
+      if (level == allowedRootLogLevel) {
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java]: MAJOR Change this comparison to use the equals method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)

File: server/sonar-process/src/main/java/org/sonar/process/Lifecycle.java [hunk 5]
```diff
@@ -101,4 +112,16 @@ public boolean equals(@Nullable Object o) {
   public int hashCode() {
     return state.hashCode();
   }
+
+  public interface LifecycleListener {
+    /**
+     * Called when a transition from state {@code from} to state {@code to} failed.
+     */
+    void failedTransition(State from, State to);
```
Comment (simonbrandhof) [inline server/sonar-process/src/main/java/org/sonar/process/Lifecycle.java]: The unique existing implementation does nothing in the method `failedTransition`. This code is useless. I suggest to remove it.

File: server/sonar-process/src/main/java/org/sonar/process/Lifecycle.java [hunk 6]
```diff
@@ -80,6 +88,9 @@ public synchronized boolean tryToMoveTo(State to) {
     if (TRANSITIONS.get(currentState).contains(to)) {
       this.state = to;
       res = true;
+      listeners.forEach(listener -> listener.successfulTransition(currentState, to));
```
Comment (simonbrandhof) [inline server/sonar-process/src/main/java/org/sonar/process/Lifecycle.java]: Missing UTs.

File: server/sonar-process/src/main/java/org/sonar/process/Lifecycle.java (hunk 5 above)
Comment (sns-seb) [inline server/sonar-process/src/main/java/org/sonar/process/Lifecycle.java]: done

**Pull request merged**